src/uwtools/yaml_file.py

```diff
@@ -0,0 +1,50 @@
+# (C) Copyright 2020-2022 UCAR
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Part of this software is developed by the Joint Center for Satellite Data Assimilation (JCSDA) together with its partners.
+#
+# (C) Copyright 2022-2022 NOAA
+import yaml
+from .nice_dict import NiceDict
+
+class YAMLFile(NiceDict):
+
+    """
+        Reads a YAML file as a NiceDict and recursively converts
+        nested dictionaries into NiceDict.
+        Provides a way of saving the dictionary issued from the yaml file, after modification or not.
+    """
+
+    def __init__(self, config_file=None, data=None):
+        super().__init__()
+        if config_file is not None:
+            with open(config_file) as f:
+                config = yaml.load(f, Loader=yaml.Loader)
+        else:
+            config = data
+        if config is not None:
+            self.update(self._configure(config))
+
+    def _configure(self, config):
```

Comment:
Why is this needed?  It looks like you're copying values one at a time from into a new structure, rather than just copying the structure.